pandas/core/strings.py

```diff
@@ -2401,11 +2401,137 @@ def rindex(self, sub, start=0, end=None):
 
     _shared_docs['ismethods'] = ("""
     Check whether all characters in each string in the Series/Index
-    are %(type)s. Equivalent to :meth:`str.%(method)s`.
+    are %(type)s.
+
+    This is equivalent to running the Python string method
+    :meth:`str.%(method)s` for each element of the Series/Index. If a string
+    has zero characters, `False` is returned for that check.
 
     Returns
     -------
-    is : Series/array of boolean values
+    Series
+        Series of boolean values with the same length as the original
+        Series/Index.
+
+    See Also
+    --------
+    Series.str.isalpha : Check whether all characters are alphabetic.
+    Series.str.isnumeric : Check whether all characters are numeric.
+    Series.str.isalnum : Check whether all characters are alphanumeric.
+    Series.str.isdigit : Check whether all characters are digits.
+    Series.str.isdecimal : Check whether all characters are decimal.
+    Series.str.isspace : Check whether all characters are whitespace.
+    Series.str.islower : Check whether all characters are lowercase.
+    Series.str.isupper : Check whether all characters are uppercase.
+    Series.str.istitle : Check whether all characters are titlecase.
+
+    Examples
+    --------
+    **Checks for Alphabetic and Numeric Characters**
+
+    >>> s1 = pd.Series(['AB', 'C12', '42', ''])
```

Comment:
We haven't done it anywhere yet, but I think it could make sense to add an index with the same values. It'd make very easy to follow the examples.

As a personal opinion, I find a bit distracting using too arbitrary examples. I'd prefer a real-world example, but as seems difficult for this case, I'd prefer something like `['one', 'one1', '1', '']`, which makes it obvious what is being shown, and don't let the users guessing why `42` and not `43`.

Reply:
Just checking I understand, do you mean having the series values and index be the same so the values are displayed side by side like below?  

```
>>> s1 = pd.Series(data=['one', 'one1', '1', ''], 
                   index=['one', 'one1', '1', ''])

>>> s1.str.isalpha()
one      True
one1    False
1       False
        False
dtype: bool
```

Reply:
It's just an idea, but yes, that's what I meant. We haven't done it in any docstring yet, afaik. But I think it makes very easy to see which value is true for each method.

Reply:
It does make comparisons easier, though I find having a blank in the last index position spoils it a bit. Could explicitly label it with 'empty string' as below?
```
>>> s1.str.isalpha()
one              True
one1            False
1               False
empty string    False
dtype: bool
```